### PR TITLE
pyDKB: v.0.3

### PR DESCRIPTION
**Pull request by mgolosova**
Comparing to 0.2:
* unified exceptions handling;
* new command line parameter (`--config`);
* custom EOM/EOP markers allowed;
* added `log()` method for pyDKB stages;
* `XXXProcessorStage` classes are replaced with `ProcessorStage` with
  adjustable input/output message types;
* clarified stages help message and execution logic.

Plus bug fixes, refactorings and minor improvements.

---

All stages using the `pyDKB.dataflow.stage` submodule are made compatible with the new version.

### CODE REVIEW

File: Utils/Dataflow/pyDKB/dataflow/stage/AbstractStage.py [hunk 1]
```diff
@@ -53,17 +70,53 @@ def add_argument(self, *args, **kwargs):
...
 
+    def config_error(self, message="Failed to read config file:"):
+        """ Output error message and exit with code 3. """
+        if self._error:
+            message = message + "\n" + str(self._error['exception'])
+        self.output_error(message)
+        sys.exit(3)
+
+    def read_config(self):
+        """ Reads stage custom config file.
```
Comment (Evildoor) [inline Utils/Dataflow/pyDKB/dataflow/stage/AbstractStage.py]: I suggest changing "Reads" to "Read" to be in line with other functions and methods in this file and pyDKB, which are described as "do X", and not "(this function/method) does X".

Comment (mgolosova) [reply Utils/Dataflow/pyDKB/dataflow/stage/AbstractStage.py]: Done: #259
Not in this PR, as it is supposed to be "merge" PR, not "merge-and-improve" one: if some improvement appears here, it won\`t get to `pyDKB` branch before `master` is merged there (again).
Yet after #259 is merged, I can (re-)merge `pyDKB` here, to make sure this change goes to `master` as well.

Comment (anastasiakaida) [reply Utils/Dataflow/pyDKB/dataflow/stage/AbstractStage.py]: Haven't found anything strange, so I suggest to approve this PR.

**Comment by mgolosova** [comment]
@anastasiakaida, I just merged here last changes from `pyDKB`; could you please re-approve?

**Comment by anastasiakaida** [comment]
@mgolosova Sure!

**Pull request merged**